### PR TITLE
Add doxygen comments to chgres_cube.

### PR DESCRIPTION
**Pull request by GeorgeGayno-NOAA**
Update chgres_cube program for doxygen.   Posted here:
https://georgegayno-noaa.github.io/UFS_UTILS/

Issue #191.

### CODE REVIEW

File: sorc/chgres_cube.fd/atmosphere.F90 [hunk 1]
```diff
@@ -1,29 +1,29 @@
- module atmosphere
+!> @file
+!! @brief Process atmospheric fields
+!!
+!! @author gayno NCEP/EMC
+!!
+!! Abstract: Process atmospheric fields:  Horizontally interpolate input
```
Comment (edwardhartnett) [inline sorc/chgres_cube.fd/atmosphere.F90:6]: Take out the "Abstract:" everywhere. Doxygen output already indicates what each section of text is, so we don't need to.

File: sorc/chgres_cube.fd/search_util.f90 [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
+!! @brief Replace undefined surface values.
+!!
+!! @author gayno NCEP/EMC
+!!
+!! Abstract: Replace undefined values with a valid value.  This can
+!!   happen for an isolated lake or island that is unresolved by
+!!   the input grid.
+!!
+!! Public Subroutines:
+!! -------------------
```
Comment (edwardhartnett) [inline sorc/chgres_cube.fd/search_util.f90:11]: Take out all instances of "-----------------------------------" or anything else that was put in as a text separator in the comments. The only acceptable separator is a single blank line.

File: sorc/chgres_cube.fd/static_data.F90 [hunk 3]
```diff
@@ -1,39 +1,40 @@
...
+!! Public Subroutines:
+!! -------------------
+!! - get_static_fields -       Driver routine to read/time interpolate
+!!                             static/climo fields on the fv3 target
+!!                             grid.
+!! - cleanup_static_fields -   Free up memory for fields in this module.
+!!
+!! Public variables:
+!! -----------------
+!! - alnsf_target_grid -                near ir black sky albedo
```
Comment (edwardhartnett) [inline sorc/chgres_cube.fd/static_data.F90:19]: This is fine for a first pass, but in a future pass we should move the documentation of the variables to the variables themselves. Then we can remove this list of public variables - doxygen constructs its own list from the code anyway.

File: sorc/chgres_cube.fd/surface.F90 [hunk 4]
```diff
@@ -1,24 +1,25 @@
...
+!!
+!! Abstract: Process surface and nst fields.  Interpolates fields from
+!!    the input to target grids.  Adjusts soil temperature according
+!!    to differences in input and target grid terrain.  Rescales
+!!    soil moisture for soil type differences between input and target
+!!    grid.  Computes frozen portion of total soil moisture.
+!!
+!! Public Subroutines:
+!! -----------------
+!! - surface_driver -         Driver routine to process surface/nst data
```
Comment (edwardhartnett) [inline sorc/chgres_cube.fd/surface.F90:14]: You can remove the list of public subroutines everywhere. Doxygen already figures that out from the code. The documentation describing each subroutine can become the first sentence of the documentation of that subroutine.

**Review by edwardhartnett — CHANGES_REQUESTED** [review]

File: sorc/chgres_cube.fd/input_data.F90 [hunk 5]
```diff
@@ -6429,6 +6422,11 @@ subroutine read_grib_soil(the_file,inv_file,vname,vname_file,dummy3d,rc)
 
  end subroutine read_grib_soil
 
+!---------------------------------------------------------------------------
```
Comment (edwardhartnett) [inline sorc/chgres_cube.fd/input_data.F90:6425]: All these lines with "--------------------------" should be removed everywhere.

File: sorc/chgres_cube.fd/search_util.f90 [hunk 6]
```diff
@@ -21,21 +17,23 @@ module search_util
...
+!! a nearby neighbor.  Undefined values are typically associated
+!! with isolated islands where there is no source data.
+!!
+!! Routine searches a neighborhood with a radius of 100 grid points.
+!! If no valid value is found, a default value is used.
 !
-! Note: This routine works for one tile of a cubed sphere grid.  It
-! does not consider valid values at adjacent faces.  That is a 
-! future upgrade.
+!! Note: This routine works for one tile of a cubed sphere grid.  It
```
Comment (edwardhartnett) [inline sorc/chgres_cube.fd/search_util.f90:30]: You can change Note: to @note and Doxygen will format it as a note...

File: sorc/chgres_cube.fd/atmosphere.F90 (hunk 1 above)
Comment (GeorgeGayno-NOAA) [reply sorc/chgres_cube.fd/atmosphere.F90]: Done.

File: sorc/chgres_cube.fd/search_util.f90 (hunk 2 above)
Comment (GeorgeGayno-NOAA) [reply sorc/chgres_cube.fd/search_util.f90]: Done.

File: sorc/chgres_cube.fd/surface.F90 (hunk 4 above)
Comment (GeorgeGayno-NOAA) [reply sorc/chgres_cube.fd/surface.F90]: Done.

File: sorc/chgres_cube.fd/input_data.F90 (hunk 5 above)
Comment (GeorgeGayno-NOAA) [reply sorc/chgres_cube.fd/input_data.F90:6425]: I use "--------------------" all through my codes to block comments.  I removed them from any doxygen comments.  But not from the general Fortran comments.

File: sorc/chgres_cube.fd/search_util.f90 (hunk 6 above)
Comment (GeorgeGayno-NOAA) [reply sorc/chgres_cube.fd/search_util.f90]: Done.

**Review by edwardhartnett — APPROVED** [review]

**Comment by GeorgeGayno-NOAA** [comment]
@edwardhartnett Should I update the `gh-pages` branch in the authoritative fork?

**Pull request merged**